lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -91,9 +97,51 @@ private function setInMemcache( $userId, $attributes ) {
 	 * @param Attribute $attribute
 	 */
 	public function setAttribute( $userId, Attribute $attribute ) {
-		$this->setInService( $userId, $attribute );
+		$this->loadAttributes( $userId );
```

Comment:
It's basically to make sure we have all attributes loaded into the instance cache (`$this->attributes)` for later on.

When we call setAttribute, we set the new value for the attribute in the service, and then in `this->attributes`.

If we haven't loaded attributes for a user yet, then the only attribute they'll have in `this->attributes` is the one that was just set. Then, if we call `getAttribute`, it'll first try and load attributes and, seeing that we already have something in `this->attributes` for the user [here](https://github.com/Wikia/app/blob/dev/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L61-L63), will return those attributes. Which means they'll only have the one attribute.